alerting/src/test/kotlin/org/opensearch/alerting/MonitorRunnerServiceIT.kt

```diff
@@ -1190,6 +1191,133 @@ class MonitorRunnerServiceIT : AlertingRestTestCase() {
         assertEquals("Incorrect search result", 2, buckets.size)
     }
 
+    fun `test execute bucket-level monitor with alias optimization - indices not skipped`() {
```

Comment:
can we add a test case with partial index consideration e.g. after the current monitor run, `5 new docs are added to index1`, `index1 is rolled over to index2`, `5 new docs added to index2`, `index2 is rolled over to index3`, `5 new  docs added to index3`,  `index3 is rolled over to index4`, `5 new  docs added to index4` & now `bucket-level monitor run` happens. So, this run considers, `index1, index2, index3 & index4`.

Reply:
added 2 test scenarios

scenario one 
12 indices in alias
1st index skipped as creation time out of time range of query contains 3 docs (3 new buckets in response)
2nd index NOT SKIPPED with creation time out of time range of query contains 3 docs (3 new buckets in response)
10 indices contains 2 docs (2 new buckets in response)
Response has only 5 buckets 

scenario 2
12 indices in alias
1st index NOT SKIPPED as creation time in time range of query contains 3 docs (3 new buckets in response)
2nd index NOT SKIPPED with creation time in time range of query contains 3 docs (3 new buckets in response)
10 indices contains 2 docs (2 new buckets in response)
Response has only 8 buckets